algopy_testing/src/algopy/op.py

```diff
@@ -0,0 +1,171 @@
+import hashlib
+from contextvars import copy_context
+from enum import Enum
+
+import algosdk
+import coincurve
+import nacl.exceptions
+import nacl.signing
+from Cryptodome.Hash import keccak
+from ecdsa import (  # type: ignore  # noqa: PGH003
+    BadSignatureError,
+    NIST256p,
+    SECP256k1,
+    VerifyingKey,
+)
+
+from algopy import Bytes, UInt64
+from algopy.utils import as_bytes
+
+
+class ECDSA(Enum):
+    Secp256k1 = 0
+    Secp256r1 = 1
+
+
+def sha256(a: Bytes | bytes, /) -> Bytes:
+    input_value = as_bytes(
+        a,
+    )
+    return Bytes(hashlib.sha256(input_value).digest())
+
+
+def sha3_256(a: Bytes | bytes, /) -> Bytes:
+    input_value = as_bytes(a)
+    return Bytes(hashlib.sha3_256(input_value).digest())
+
+
+def keccak256(a: Bytes | bytes, /) -> Bytes:
+    input_value = as_bytes(a)
+    hashed_value = keccak.new(data=input_value, digest_bits=256)
+    return Bytes(hashed_value.digest())
+
+
+def sha512_256(a: Bytes | bytes, /) -> Bytes:
+    input_value = as_bytes(a)
+    hash_object = hashlib.new("sha512_256", input_value)
```

Comment:
this isn't cross-platform, it relies on OpenSSL (see CPython #71021)

since it looks like there's already a dependency, on Cryptodome, you can use that instead for cross-platform support:

```python3
from Cryptodome.Hash import SHA512

sha = SHA512.new(truncate="256")
sha.update(input_value)
return sha.digest()
```